hudi-common/src/main/java/org/apache/hudi/metadata/HoodieMetadataPayload.java

```diff
@@ -155,7 +156,7 @@ public class HoodieMetadataPayload implements HoodieRecordPayload<HoodieMetadata
   /**
    * HoodieMetadata secondary index payload field ids
    */
-  public static final String SECONDARY_INDEX_FIELD_RECORD_KEY = "recordKey";
+  public static final String SECONDARY_INDEX_RECORD_KEY_SEPARATOR = "$";
```

Comment:
the key for the secondary index record has to be a string and it is in the format `secondaryKey$recordKey`. Even though the record key is extracted using lastIndexOf($), still it won;t work if the record key itself contains the separator. Is there a better way to encode the key?

Reply:
i was thinking of doing base64 encoding at first but then escaping is just as efficient. Ran a local [benchmark](https://gist.github.com/codope/b1c73abed748d77c0b4db974d835f9da), not much difference between the two.